tests/Unit/SDK/Trace/SpanTest.php

```diff
@@ -669,6 +672,51 @@ public function test_dropping_events(): void
 
     // endregion SDK
 
+    /**
+    * @group trace-compliance
+    */
+    public function test_set_attributes_merges_attributes(): void
+    {
+        $span = $this->createTestRootSpan();
+
+        $attributes = new Attributes([
+            'string' => 'str_val',
+            'empty_key' => '',
+            'str_array' => ['f', 'b'],
+        ]);
+
+        $span->setAttribute('str_array', ['a', 'b']);
+        $span->setAttribute('string', 'str');
+
+        $span->setAttributes($attributes);
+        $span->end();
+
+        $attributes = $span->toSpanData()->getAttributes();
+        $this->assertSame('str_val', $attributes->get('string'));
+        $this->assertSame('', $attributes->get('empty_key'));
+        $this->assertSame(['f', 'b'], $attributes->get('str_array'));
+    }
+
+    /**
+     * @group trace-compliance
+     */
+    public function test_add_event_order_preserved(): void
+    {
+        $span = $this->createTestRootSpan();
+        $span->addEvent('a');
+        $span->addEvent('b');
+        $span->addEvent('c', new Attributes(['key' => 2]));
+
+        $span->end();
+
+        $events = $span->toSpanData()->getEvents();
+        $c = 0;
+
+        $this->assertEvent($events[$c++], 'a', new Attributes(), self::START_EPOCH);
```

Comment:
I think this would be more readable if you just provided integer values.

Reply:
Agreed! I've pushed [a9270a4](https://github.com/open-telemetry/opentelemetry-php/pull/669/commits/a9270a42fd3c5cea90540c2c8ebcb98a0befaa25) to improve readability here. Can you take a look?